### PR TITLE
HTTP2 client support

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

This adds HTTP2 support in curl for depends platforms.
Windows would need some love, but libnghttp2 doesn't need any dependencies we don't already have

## How Has This Been Tested?

only on linux, via youtube plugin

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/filesystem/CurlFile.cpp [hunk 1]
```diff
@@ -626,6 +626,10 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
   // Setup allowed TLS/SSL ciphers. New versions of cURL may deprecate things that are still in use.
   if (!m_cipherlist.empty())
     g_curlInterface.easy_setopt(h, CURLOPT_SSL_CIPHER_LIST, m_cipherlist.c_str());
+
+  // enable HTTP2 support. default: CURL_HTTP_VERSION_1_1. Curl >= 7.62.0 defaults to CURL_HTTP_VERSION_2TLS
+  g_curlInterface.easy_setopt(h, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_0);
```
Comment (pkerling) [inline xbmc/filesystem/CurlFile.cpp]: No support for building with older curl?

What is the problem with `CURL_HTTP_VERSION_2TLS`?

Comment (Rechi) [reply xbmc/filesystem/CurlFile.cpp]: `CURL_HTTP_VERSION_2_0` was added in version 7.33.0 (14th October 2013). So why would you want to support even older versions?

Comment (pkerling) [reply xbmc/filesystem/CurlFile.cpp]: > CURL_HTTP_VERSION_2_0 was added in version 7.33.0 (14th October 2013). So why would you want to support even older versions?

Crazy how time flies. To me HTTP2 still seemed to be a totally recent technology :-) Disregard then.

**Comment by wsnipex** [comment]
I'm unsure if we should consider this for v18. It's very late in the cycle, but otoh rather low impact.

**Comment by MartijnKaijser** [comment]
Maybe for point release?

**Comment by wsnipex** [comment]
I thought about that as well, so fine for me

**Comment by afedchin** [comment]
> Windows would need some love, but libnghttp2 doesn't need any dependencies we don't already have

vcpkg proj has nhttp2 port, so windows isn't an issue but need some time to compile

**Comment by wsnipex** [comment]
@afedchin any news?

**Comment by afedchin** [comment]
@wsnipex about what? do you need compiled libs? @Rechi can help you, he re-compiled curl recently iirc.

**Comment by wsnipex** [comment]
hm ok, I interpreted your last comment as if you would compile it.
just for completeness: I don't need the windows port, I can merge this as is(without windows support)

**Comment by MartijnKaijser** [comment]
@Rechi can you compile new libs or do we merge it without windows updated?

**Comment by Rechi** [comment]
Not going to touch windows depends for this feature in a point release.

File: xbmc/filesystem/CurlFile.cpp (hunk 1 above)
Comment (wsnipex) [reply xbmc/filesystem/CurlFile.cpp]: @Rechi could you please test if windows is ok with this without nghttp2 or if I need to make this posix only?

Comment (Rechi) [reply xbmc/filesystem/CurlFile.cpp]: updating & installing addons and playing videos from http & https still works

**Comment by wsnipex** [comment]
Jenkins build this please

**Comment by wsnipex** [comment]
jenkins build and merge

**Pull request merged**